html_sanitizer/sanitizer.py

```diff
@@ -34,14 +56,20 @@ def normalize_overall_whitespace(html):
         "\r",
         "&#13;",
         "&#xd;",
-        "\xa0",
-        "&nbsp;",
-        "&#160;",
-        "&#xa0",
     ]
+    if not keep_typographic_whitespace:
+        # non-breaking space representations
+        whitespace += [
+            "\xa0",
+            "&nbsp;",
+            "&#160;",
+            "&#xa0;",
+        ]
+    if whitespace_re is None:
+        whitespace_re = r"\s+"
     for ch in whitespace:
         html = html.replace(ch, " ")
-    html = re.sub(r"(?u)\s+", " ", html)
+    html = re.sub(whitespace_re, " ", html)
```

Comment:
Aren't we losing the normalization of unicode spaces here?

Reply:
Yeah, you're right at least for Python 2 (shouldn't make a difference in Python 3 as strings are handled as Unicode anyway). I'm inserting the Unicode flag into the regex for Python 2 now.